pkg/controllers/rollout/controller.go

```diff
@@ -160,6 +169,54 @@ func (r *Reconciler) fetchLatestResourceSnapshot(ctx context.Context, crpName st
 	return latestResourceSnapshotName, nil
 }
 
+// waitForResourcesToCleanUp checks if there are any cluster that has a binding that is both being deleted and another one that needs rollout.
+// We currently just wait for those cluster to be cleanup so that we can have a clean slate to start compute the rollout plan.
+// TODO (rzhang): group all bindings pointing to the same cluster together when we calculate the rollout plan so that we can avoid this.
+func waitForResourcesToCleanUp(allBindings []*fleetv1beta1.ClusterResourceBinding, crp *fleetv1beta1.ClusterResourcePlacement) (bool, error) {
+	crpObj := klog.KObj(crp)
+	deletingBinding := make(map[string]bool)
+	bindingMap := make(map[string]*fleetv1beta1.ClusterResourceBinding)
+	// separate deleting bindings from the rest of the bindings
+	for _, binding := range allBindings {
+		if !binding.DeletionTimestamp.IsZero() {
+			deletingBinding[binding.Spec.TargetCluster] = true
+			klog.V(2).InfoS("Found a binding that is being deleted", "clusterResourcePlacement", crpObj, "binding", klog.KObj(binding))
+		} else {
+			if _, exist := bindingMap[binding.Spec.TargetCluster]; !exist {
+				bindingMap[binding.Spec.TargetCluster] = binding
+			} else {
+				return false, controller.NewUnexpectedBehaviorError(fmt.Errorf("the same cluster `%s` has bindings `%s` and `%s` pointing to it",
+					binding.Spec.TargetCluster, bindingMap[binding.Spec.TargetCluster].Name, binding.Name))
+			}
+		}
+	}
+	// check if there are any cluster that has a binding that is both being deleted and scheduled
+	for cluster, binding := range bindingMap {
+		// check if there is a  deleting binding on the same cluster
+		if deletingBinding[cluster] {
+			klog.V(2).InfoS("Find a binding assigned to a cluster with another deleting binding", "clusterResourcePlacement", crpObj, "binding", binding)
+			if binding.Spec.State == fleetv1beta1.BindingStateBound {
+				// the rollout controller won't move a binding from unscheduled to bound if there is a deleting binding on the same cluster.
```

Comment:
Hi Ryan! A nit: the comment should be `... from scheduled to bound ...` I assume?